### ISSUE TITLE
Wrong color accent for quick setting checkboxes on GMaps

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#9576]
Buttons are orange, checkboxes however are blue.

![Screenshot_20201223-122722_cgeo](https://user-images.githubusercontent.com/949669/102991915-818e8780-451a-11eb-99f1-46d82bc1c343.jpg)

Related to #9373 and #8783

**Comment by MagpieFourtyTwo** [comment]
This is pretty strange - in even two ways:
1. the checkboxes over here look completely different, i. e. orange and not round but square, thus perfectly fine
2. we already had a similar issue, which has already been fixed by @moving-bits, IIRC, so are you perhaps using an old version?

**Comment by MagpieFourtyTwo** [comment]
Ah, I'm currently using 2020.12.26-NB-a2423d5

**Comment by Lineflyer** [comment]
I think it could be more related to your Android version...or you did not use GMaps.

**Comment by MagpieFourtyTwo** [comment]
The boxes look the same, no matter if I use OSM or GMap or GSat.
Android is 10.1.0.171, Huawei.

**Comment by moving-bits** [comment]
I can only guess here, as the dialog already activates a `ContextThemeWrapper` which sets the theme to our `DialogAlert` theme, which sets the `colorAccent` to the desired value.

But I've stumbled already earlier about the fact, that some themes work with `colorAccent` where others use `android:colorAccent` (or, to be more precise: checkboxes in some themes use it), so I'll prepare a test PR which sets `android:colorAccent` to `DialogAlert` theme - let's see if that helps.

**Comment by Lineflyer** [comment]
@moving-bits  #9585 did not change anything related to this issue for me.
Please note, that besides the mentioned quick settings, e.g. also the map selection on GMaps has blue accents (and black background) on my device. I guess the complete styling needs to be equalized to OSM-Maps (see #9373) as some parents are siginificantly different.

**Comment by Lineflyer** [comment]
This is how GMaps in c:geo looks for me (similar to settings).
Android 11, Samsung S20, 2020.12.26-NB1

![Screenshot_20201226-221915_cgeo](https://user-images.githubusercontent.com/949669/103159329-8710f980-47c8-11eb-8090-cb59d8da265d.jpg)
![Screenshot_20201226-221909_cgeo](https://user-images.githubusercontent.com/949669/103159330-88422680-47c8-11eb-98a2-3381b122de98.jpg)
![Screenshot_20201226-221902_cgeo](https://user-images.githubusercontent.com/949669/103159331-88422680-47c8-11eb-83f9-84f6142f2cc4.jpg)

**Comment by Lineflyer** [comment]
Addition:
As soon as I switch the device to c:geo white theme, the accent is OK. See screenshots below.
Also the top bar of my device is no longer blue and map title bar background is grey instead of black.

So whatever is applied to white theme for coloring and menu style probably needs to be copied to the black theme (except the menu background color of course).

![Screenshot_20201226-222438_cgeo](https://user-images.githubusercontent.com/949669/103159409-701ed700-47c9-11eb-8a32-93701ce20c92.jpg)
![Screenshot_20201226-222433_cgeo](https://user-images.githubusercontent.com/949669/103159410-70b76d80-47c9-11eb-9be2-0af7ba0e4128.jpg)
![Screenshot_20201226-222451_cgeo](https://user-images.githubusercontent.com/949669/103159411-71500400-47c9-11eb-99c2-fa8588149fb3.jpg)
![Screenshot_20201226-222444_cgeo](https://user-images.githubusercontent.com/949669/103159413-71500400-47c9-11eb-8c31-544564f78ed9.jpg)

**Comment by moving-bits** [comment]
This is truely strange. The relevant section in our theme definition file for Google Maps looks like this:

```
<style name="cgeo_gmap" parent="@android:style/Theme.DeviceDefault">
    <item name="text_color">@color/text_dark</item>
    <item name="button">@drawable/action_button_dark</item>
    <item name="colorAccent">@color/colorAccent</item>
    <item name="android:colorAccent">@color/colorAccent</item>
</style>

<style name="cgeo_gmap_light" parent="@android:style/Theme.DeviceDefault.Light.DarkActionBar">
    <item name="text_color">@color/text_light</item>
    <item name="button">@drawable/action_button_light</item>
    <item name="colorAccent">@color/colorAccent</item>
    <item name="android:colorAccent">@color/colorAccent</item>
</style>
```

So both have the same individual definitions, and they seem to inherit from the same theme set `Theme.DeviceDefault`.

**Comment by Lineflyer** [comment]
I have no detailed idea about theme definitions.
But if the parent is different in both styles do the references (e.g. @color/...) not also target potentially different definitions in its parent although their name is the same?

**Comment by moving-bits** [comment]
parent in both cases is `Theme.DeviceDefault`, and then for the light theme there are some additions for the light theme (which is ok). So both themes should have the same base to which we set our `colorAccent`.
It looks like some controls themself are based on a different theme than, so I'm looking for that base. Hm :-(

**Comment by Lineflyer** [comment]
> It looks like some controls themself are based on a different theme than, so I'm looking for that base. Hm :-(

Up to my experience it has to be somewhere in the GMaps world and a bit specific to Android 9 and above (lower Android versions seem to look OK). Is there any inherited style which comes from the map itself?

**Comment by moving-bits** [comment]
After the recent upgrade to Android 11 there are even more places which do not show the `colorAccent` on my physical mobile, even though they have worked before. It's somewhat frustrating...
Maybe it's time to stop fine-tuning here (and related issues regarding `colorAccent`) and wait for #9373...

**Comment by Lineflyer** [comment]
Shall I drive you totally crazy:
On my Android 5 device it was the same blue/orange problem before.
Now checkboxes on Gmaps and in Settings are **GREEN**
Checkboxes on OSM are orange

No idea why...

**Comment by Lineflyer** [comment]
Reason for green could be, that it is now somehow inherited from system itself as the android system settings e.g. also have green checkboxes.

**Comment by Lineflyer** [comment]
Here we go:
Samsung S4, Android 5, GMaps (same in settings)

The only consistent thing compared to my initial problem on Samsung S20, Android 11 is, that the wrong color seems to be the color used in system checkboxes and radio buttons (green on S4, light blue on S20). Maybe that helps.



![Screenshot_2021-01-27-23-35-15](https://user-images.githubusercontent.com/949669/106063556-a87f4100-60f8-11eb-8c48-891602a7f8c4.png)
![Screenshot_2021-01-27-23-35-24](https://user-images.githubusercontent.com/949669/106063559-a917d780-60f8-11eb-8325-accad4b3d87a.png)

**Comment by moving-bits** [comment]
Needs to be re-checked after merging #10049

**Comment by fm-sys** [comment]
@Lineflyer could you retest this issue at your Samsung device?

**Comment by Lineflyer** [comment]
Fixed on all my mentioned devices with 2021.04.02-NB!!
